### DISCUSSION TITLE
How to execute some python code with some pre-populated parameters from JavaScript?

### DISCUSSION THREAD

**Discussion by wlach** [pyodide/pyodide#2322] [Q&A] [answered]
I'd like to be able to inject some variables from my JavaScript context into python as its global namespace, but I can't figure out how to do it offhand.

There's this chunk from the documentation:

https://pyodide.org/en/stable/usage/faq.html#how-can-i-execute-code-in-a-custom-namespace

However, that defines how to create a *blank* namespace for python to write in. Also, it seems not to currently be working (see: https://github.com/pyodide/pyodide/issues/2321). Is there a way to be able to define some variables in advance (I guess creating a python dictionary from JavaScript?)? Something like this:

```js
let my_namespace = pyodide.globals.dict({x: 2}); // doesn't actually work
pyodide.runPython(`y = x ** x`, my_namespace);
my_namespace.y; // ==> 4
```

**Comment by hoodmane** [comment, marked as answer]
I think what you want is something like:
```js
let my_namespace = pyodide.toPy({x : 2, y : [1,2,3]});
pyodide.runPython(`
	assert x == y[1]
	z = x ** x
`, my_namespace);
my_namespace.get("z"); // 4
```
